extensions/wikia/AdEngine/js/video/player/porvata/floater.js

```diff
@@ -0,0 +1,175 @@
+			}
+		}
+
+		function resetDimensions(element, params) {
+			updateDimensions(element, params.width, params.height);
+		}
+
+		function resetImage(image, marginTop) {
+			if (image) {
+				image.removeAttribute('width');
+				image.style.marginTop = marginTop;
+			}
+		}
+
+		function enableFloating(elements) {
+			var width = ((win.innerWidth - elements.background.offsetWidth) / 2) - floatingVideoPadding,
+				height = width;
+
+			elements.top.classList.toggle(activeFloatingCssClass);
+			updateDimensions(elements.top, width, height);
+			updateDimensions(elements.iframe, width, height);
+			updateDimensions(elements.imageContainer, width, height);
+			updateImage(elements.image, width, height);
+			elements.video.resize(width, height);
+		}
+
+		function disableFloating(elements, params, imageMarginTop) {
+			elements.top.classList.toggle(activeFloatingCssClass);
+			resetAdContainer(elements.top);
+			resetDimensions(elements.iframe, params);
+			resetDimensions(elements.imageContainer, params);
+			resetImage(elements.image, imageMarginTop);
+			elements.video.resize(params.width, params.height);
+		}
+
+		function enableFloatingOn(video, params, onClose) {
+			var top = doc.getElementById('WikiaTopAds'),
+				threshold = 100,
+				scrollYOffset = top.offsetTop + top.offsetHeight + threshold,
+				imageMarginTop,
+				floatingContext = {
+					onClose: onClose
+				},
+				imageContainer = params.container.parentElement.querySelector('#image'),
+				elements = {
+					top: top,
+					background: doc.getElementById('WikiaPageBackground'),
+					iframe: top.querySelector('iframe'),
+					imageContainer: imageContainer,
+					image: imageContainer ? imageContainer.querySelector('img') : null,
+					video: video
+				};
+
+			if (!floatingContext.closeButton) {
+				floatingContext.closeButton = createCloseButton();
+				floatingContext.closeButton.addEventListener('click',
+					createOnCloseListener(elements, params, floatingContext, imageMarginTop));
+
+				elements.top.appendChild(floatingContext.closeButton);
+			}
+
+			floatingContext.scrollListener = function () {
+				if (win.scrollY > scrollYOffset) {
+					if (!floatingContext.floating) {
+						if (imageMarginTop === undefined && elements.image) {
+							imageMarginTop = elements.image.style.marginTop;
+						}
+
+						enableFloating(elements);
+
+						if (floatingContext.closeButton) {
+							floatingContext.closeButton.classList.remove('hidden');
+						}
+
+						floatingContext.floating = true;
+					}
+				} else {
+					if (floatingContext.floating) {
+						disableFloating(elements, params, imageMarginTop);
+
+						if (floatingContext.closeButton) {
+							floatingContext.closeButton.classList.add('hidden');
+						}
+
+						if (floatingContext.videoEnded) {
+							win.removeEventListener('scroll', floatingContext.scrollListener);
+
+							deleteCloseButton(elements, floatingContext);
+						}
+
+						floatingContext.floating = false;
+					}
+				}
+			};
+
```

Comment:
I don't like this scroll listener created here. It's very long and makes `enableFloatingOn` a 60line function. What about moving its creation to a separate function and/or add comments so the person that is reading that doesn't need to think that much ie. "WikiaTopAds is outside of the viewport" or what means that `imageMarginTop === undefined && elements.image`